engine/render/src/render/render_private.h

```diff
@@ -39,26 +39,26 @@ namespace dmRender
 
     struct Sampler
     {
-        dmhash_t m_NameHash;
-        int32_t  m_Location : 24;
-        int32_t  m_Unit     : 8;
-
+        dmhash_t                  m_NameHash;
+        dmGraphics::TextureType   m_Type;
         dmGraphics::TextureFilter m_MinFilter;
         dmGraphics::TextureFilter m_MagFilter;
-        dmGraphics::TextureWrap m_UWrap;
-        dmGraphics::TextureWrap m_VWrap;
-
-        float m_MaxAnisotropy;
+        dmGraphics::TextureWrap   m_UWrap;
+        dmGraphics::TextureWrap   m_VWrap;
+        float                     m_MaxAnisotropy;
+        int32_t                   m_Location       : 24;
+        int32_t                   m_UnitValueCount : 8;
 
-        Sampler(int32_t unit)
+        Sampler()
             : m_NameHash(0)
-            , m_Location(-1)
-            , m_Unit(unit)
```

Comment:
We never used this value, the unit is the same as the array index of the materials sampler array